test/components/views/user-onboarding/UserOnboardingPage-test.tsx

```diff
@@ -0,0 +1,101 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from "react";
+import { act, render, RenderResult } from "@testing-library/react";
+
+import { filterConsole, stubClient } from "../../../test-utils";
+import { UserOnboardingPage } from "../../../../src/components/views/user-onboarding/UserOnboardingPage";
+import { MatrixClientPeg } from "../../../../src/MatrixClientPeg";
+import SdkConfig from "../../../../src/SdkConfig";
+
+jest.mock("../../../../src/components/structures/EmbeddedPage", () => ({
+    __esModule: true,
+    default: jest.fn().mockImplementation(({ url }) => <div>{url}</div>),
+}));
+
+jest.mock("../../../../src/components/structures/HomePage", () => ({
+    __esModule: true,
+    default: jest.fn().mockImplementation(() => <div>home page</div>),
+}));
+
+describe("UserOnboardingPage", () => {
+    let restoreConsole: () => void;
+
+    const renderComponent = async (): Promise<RenderResult> => {
+        const renderResult = render(<UserOnboardingPage />);
+        await act(async () => {
+            jest.runAllTimers();
+        });
+        return renderResult;
+    };
+
+    beforeAll(() => {
+        restoreConsole = filterConsole(
+            // unrelated for this test
+            "could not update user onboarding context",
+        );
+    });
+
+    beforeEach(() => {
+        stubClient();
+        jest.useFakeTimers();
+    });
+
+    afterEach(() => {
+        jest.useRealTimers();
+    });
+
+    afterAll(() => {
+        restoreConsole();
+    });
+
+    describe("when the user registered before the cutoff date", () => {
+        beforeEach(() => {
+            jest.spyOn(MatrixClientPeg, "userRegisteredAfter").mockReturnValue(false);
+        });
+
+        it("should render the home page", async () => {
+            expect((await renderComponent()).queryByText("home page")).toBeInTheDocument();
+        });
+    });
+
+    describe("when the user registered after the cutoff date", () => {
+        beforeEach(() => {
+            jest.spyOn(MatrixClientPeg, "userRegisteredAfter").mockReturnValue(true);
+        });
+
+        describe("and there is an explicit home page configured", () => {
+            beforeEach(() => {
+                jest.spyOn(SdkConfig, "get").mockReturnValue({
+                    embedded_pages: {
+                        home_url: "https://example.com/home",
+                    },
+                });
+            });
+
+            it("should render the configured page", async () => {
+                expect((await renderComponent()).queryByText("https://example.com/home")).toBeInTheDocument();
+            });
+        });
+
+        describe("and there is no home page configured", () => {
+            it("should render the onboarding", async () => {
+                expect((await renderComponent()).queryByTestId("user-onboarding-list")).toBeInTheDocument();
```

Comment:
If possible, can we avoid the test ID and instead look out for a text from the onboarding view? Like for instance here with `queryByText('Complete these to get the most out of')` (untested).

Reply:
Sure, but just recognized this test still fails. While writing I skipped the others and this one passed. Any idea what's wrong?

Reply:
@GoodGuyMarco try adding this to `afterEach`: `jest.restoreAllMocks();` to clean up the spies.

Reply:
Works, thanks!